src/test/java/bio/terra/service/snapshot/SnapshotConnectedTest.java

```diff
@@ -635,6 +635,98 @@ public void testExcludeLockedFromSnapshotFileLookups() throws Exception {
         connectedOperations.getDatasetExpectError(datasetRefSummary.getId(), HttpStatus.NOT_FOUND);
     }
 
+    @Test
+    public void testSnapshotArrayFileRefLookups() throws Exception {
+        // create a dataset
+        DatasetSummaryModel datasetRefSummary = createTestDataset("simple-with-array-filerefs-dataset.json");
```

Comment:
I'm sure this is a complete test, but a more minimal test could be written that verifies that the `arrayOf` flag is correctly preserved in the copied column. Such as a test that calls `conjureSnapshotTablesFromRowIds()` directly.

Reply:
true and simpler is usually better--do you think it makes sense to do the switch?

Reply:
As much as I'd like to see something like that, I don't know how hard it would be to write. I just took another look at the code in `SnapshotService`, and if I were doing this, I would probably refactor the two "conjure" methods first, and then write a unit test on the refactored out method, which would only transform a `datasetColumn` to a `snapshotTable`.

Reply:
Or, a less ambitious refactor would be a method like:
```

    private void populateMapFromColumn(List<Column> columnList, List<SnapshotMapColumn> mapColumnList, Column datasetColumn) {
        Column snapshotColumn = new Column()
            .name(datasetColumn.getName())
            .arrayOf(datasetColumn.isArrayOf());
        SnapshotMapColumn snapshotMapColumn = new SnapshotMapColumn()
            .fromColumn(datasetColumn)
            .toColumn(snapshotColumn);
        columnList.add(snapshotColumn);
        mapColumnList.add(snapshotMapColumn);
    }
```
This method could be used from both conjure methods can could be tested independently of the rest of the code.

I think more refactoring could be done, as both loops that call this are very similar, but that would require more work.

Reply:
Here is a more complete refactor of the code in `SnapshotService`. I didn't run or test this so it may or may not work.

```

    private void populateMapFromColumn(List<Column> columnList, List<SnapshotMapColumn> mapColumnList, Column datasetColumn) {
        Column snapshotColumn = new Column()
            .name(datasetColumn.getName())
            .arrayOf(datasetColumn.isArrayOf());
        SnapshotMapColumn snapshotMapColumn = new SnapshotMapColumn()
            .fromColumn(datasetColumn)
            .toColumn(snapshotColumn);
        columnList.add(snapshotColumn);
        mapColumnList.add(snapshotMapColumn);
    }

    private void conjureSnapshotTablesFromDatasetTablesWithFilter(Snapshot snapshot,
                                                                  SnapshotSource snapshotSource,
                                                                  Predicate<Column> filterColumn) {
        // TODO this will need to be changed when we have more than one dataset per snapshot (>1 contentsModel)
        // for each dataset table specified in the request, create a table in the snapshot with the same name
        Dataset dataset = snapshotSource.getDataset();
        List<SnapshotTable> tableList = new ArrayList<>();
        List<SnapshotMapTable> mapTableList = new ArrayList<>();

        for (DatasetTable datasetTable : dataset.getTables()) {
            List<Column> columnList = new ArrayList<>();
            List<SnapshotMapColumn> mapColumnList = new ArrayList<>();

            // for each dataset column specified in the request, create a column in the snapshot w the same name & array
            for (Column datasetColumn : datasetTable.getColumns()) {
                populateMapFromColumn(columnList, mapColumnList, datasetColumn);
            }

            // create snapshot tables & mapping with the proper dataset name and columns
            SnapshotTable snapshotTable = new SnapshotTable(datasetTable.getName(), columnList);
            tableList.add(snapshotTable);

            mapTableList.add(new SnapshotMapTable()
                .fromTable(datasetTable)
                .toTable(snapshotTable)
                .snapshotMapColumns(mapColumnList));
        }
        // set the snapshot tables and mapping
        snapshot.snapshotTables(tableList);
        snapshotSource.snapshotMapTables(mapTableList);
    }

    private void conjureSnapshotTablesFromRowIds(SnapshotRequestRowIdModel requestRowIdModel,
                                                Snapshot snapshot,
                                                SnapshotSource snapshotSource) {
        // table name -> set of columns in table
        Map<String, Set<String>> requestTableLookup = requestRowIdModel.getTables()
            .stream()
            .collect(Collectors.toMap(SnapshotRequestRowIdTableModel::getTableName,
                table -> new HashSet<>(table.getColumns())));

        conjureSnapshotTablesFromDatasetTablesWithFilter(snapshot, snapshotSource,
            column -> {
                Set<String> columns = requestTableLookup.get(column.getTable().getName());
                return columns != null && columns.contains(column.getName());
            }
        );
    }
```

Reply:
love it--adding as a follow on!